src/main/java/org/fenixedu/academicextensions/services/registrationhistory/RegistrationHistoryReport.java

```diff
@@ -347,6 +350,11 @@ public LocalDate getLastRegistrationStateDate() {
         return state == null ? null : state.getStateDate().toLocalDate();
     }
 
+    public DateTime getLastEnrolmentUpdateDateTime() {
+        return registration.getCurriculumLineLogsSet().stream().filter(log -> log instanceof EnrolmentLog)
+                .max(Comparator.comparing(CurriculumLineLog::getDateDateTime)).map(log -> log.getDateDateTime()).orElse(null);
```

Comment:
ths solution talked before, of swapping the order of the comparator and the map didn't work?